gflownet/envs/sequence.py

```diff
@@ -0,0 +1,470 @@
+"""
+Classes to represent sequence-like environments
+Particularly AMP and DNA
+"""
+from typing import List, Tuple
+import itertools
+import numpy as np
+from gflownet.envs.base import GFlowNetEnv
+import itertools
+from polyleven import levenshtein
+import numpy.typing as npt
+from torchtyping import TensorType
+import torch
+import matplotlib.pyplot as plt
+import torch.nn.functional as F
+
+
+class Sequence(GFlowNetEnv):
+    """
+    Anti-microbial peptide sequence environment
+
+    Attributes
+    ----------
+    max_seq_length : int
+        Maximum length of the sequences
+
+    min_seq_length : int
+        Minimum length of the sequences
+
+    nalphabet : int
+        Number of letters in the alphabet
+
+    state : list
+        Representation of a sequence (state), as a list of length max_seq_length where
+        each element is the index of a letter in the alphabet, from 0 to (nalphabet -
+        1).
+
+    done : bool
+        True if the sequence has reached a terminal state (maximum length, or stop
+        action executed.
+
+    func : str
+        Name of the reward function
+
+    n_actions : int
+        Number of actions applied to the sequence
+
+    proxy : lambda
+        Proxy model
+    """
+
+    def __init__(
+        self,
+        corr_type,
+        max_seq_length=50,
+        min_seq_length=1,
+        # Not required in env. But used in config_env in MLP. TODO: Find a way out
+        n_alphabet=20,
+        min_word_len=1,
+        max_word_len=1,
+        special_tokens=None,
+        **kwargs,
+    ):
+        self.min_seq_length = min_seq_length
+        self.max_seq_length = max_seq_length
+        self.min_word_len = min_word_len
+        self.max_word_len = max_word_len
+        self.corr_type = corr_type
```

Comment:
To be revised

Reply:
Actually, this does not need to be included as an argument in any case, because it should be part of kwargs